Modules/rat_facts.py

```diff
@@ -0,0 +1,187 @@
+
+        Returns: True if it exists, False if it exists
+
+        """
+        result = await self.dbm.select_rows("fact", "AND", {"lang": lang, "name": name})
+        return len(result) > 0
+
+    async def set_fact(self, fact: Fact) -> bool or None:
+        """
+        sets a fact
+        Args:
+            fact (Fact): the fact to push into the DB
+        Returns: True on success, False if it failed
+
+        """
+        try:
+            await self.dbm.insert_row("fact",
+                                      {"name": fact.name,
+                                       "lang": fact.lang,
+                                       "message": fact.message,
+                                       "author": fact.author},
+                                      ("name", "lang",)
+                                      )
+        except ValueError:
+            return False
+        else:
+            time = datetime.datetime.now(tz=datetime.timezone.utc)
+            self.dbm._execute(r"INSERT INTO fact_timestamps (name, lang, last_modified) "
+                              r"VALUES("
+                              f"'{fact.name}', '{fact.lang}', ?)"
+                              r"ON CONFLICT (name, lang)"
+                              r"DO UPDATE SET last_modified = ?",
+                              time, time  # time is doubled on purpose
+                              )
+            return True
+
+    async def delete_fact(self, name: str, lang: str) -> bool or None:
+        """
+        delete's the given fact
+        Args:
+            name: name of the fact to delete
+            lang: language to delete
+
+        Returns: True on success, False if it failed and None if the DBM is disabled
+
+        """
+        if not self.dbm.enabled:
+            return None
+        try:
+            await self.dbm.delete_row("fact", "AND", {"name": name, "lang": lang})
+            return True
+        except ValueError:
+            return False
+
+    async def handle_fact(self, context: Context) -> bool:
+        """
+
+        Responds with the message of the fact, should a match be found and not listed as an
+        internal fact.
+
+        Args:
+            context: the context as passed by rat_command
+
+        Returns: True if it was a fact, False if not. Will also return False,
+            should the DBM be disabled.
+
+        """
+        if not self.dbm.enabled:
+            context.reply("Sadly, Facts are currently not available. Please check back later.")
+            return False
+
+        regex = re.compile(r"!(?P<name>[a-zA-Z\d]+)-(?P<lang>[a-zA-Z]{2})|!(?P<name2>[a-zA-Z\d]+)")
+        result = re.match(regex, context.words[0])
+
+        if result is None:
+            return False
+
+        name = result.group("name")
+        lang = None
+
+        if not name:
+            name = result.group("name2")
+        else:
+            lang = result.group("lang")
+
+        if not lang:
```

Comment:
If this is intended to be a `None` check then `None` should be checked for explicitly.
(Falsy check where a none check was expected)

Reply:
fixed